### PR TITLE
fix(tab): make tab respect radio groups (fix #207)

### LINKED ISSUE #207: tab with radio buttons
just discovered an issue w/ `tab()`.  it does not account for radio buttons properly and needs to more closely resemble the browser in this respect.

### PR DESCRIPTION
**Pull request by nickmccurdy**
**What**: make tab respect radio groups

**Why**: fixes bugs / inaccurate simulation with `tab()` naively focusing all radio buttons

**How**: ignore radio buttons that aren't checked or at the ends within their named groups

**Checklist**:

- [ ] Documentation
- [x] Tests
- [ ] Typings N/A
- [x] Ready to be merged
      

The algorithm I used to prune irrelevant radio buttons is a bit ugly and hacky because of all the edge cases, so I'd appreciate any ideas to improve it. We also might need to carefully document this change as it could cause tests to fail if they're relying on the incorrect focusing behavior (though changing incorrect functionality to fix a bug is technically not a breaking change in semver).

### CODE REVIEW

**Comment by nickmccurdy** [comment]
@benmonro I'm hoping this is what you meant the problem was in #207, if not let's update that issue any I may give it another try.

File: src/index.js [hunk 1]
```diff
@@ -408,15 +408,39 @@ function tab({shift = false, focusTrap = document} = {}) {
...
+      } else if (el.checked) {
+        prunedElements.splice(replacedIndex, 1)
+        prunedElements.push(el)
+      }
+    } else {
+      prunedElements.push(el)
+    }
+  }
+
+  if (shift) prunedElements.reverse()
```
Comment (benmonro) [inline src/index.js:434]: Is it already reversed since you reversed the ordered elements above?

Comment (nickmccurdy) [reply src/index.js:434]: Yes. I'm reversing it twice (before and after the pruning) when using `shift` because I couldn't figure out a clean way to make the pruning logic reverse directions without duplicating it.

**Comment by nickmccurdy** [comment]
Sorry I force pushed this to clean up the commit message, not realizing you reviewed. If you're okay with this for now we can merge, unless you want the behavior documented somewhere first.

File: src/index.js (hunk 1 above)
Comment (benmonro) [reply src/index.js:434]: Nope that's cool just checking
